src/lib/core/hooks/checkedLegendItemsStatus.ts

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from 'react';
+import { LegendItemsProps } from '@veupathdb/components/lib/components/plotControls/PlotLegend';
+import { PromiseHookState } from './promise';
+import { BarplotDataWithStatistics } from '../components/visualizations/implementations/BarplotVisualization';
+import { HistogramDataWithCoverageStatistics } from '../components/visualizations/implementations/HistogramVisualization';
+
+/**
+ * A custom hook to preserve the status of the checked legend items
+ */
+
+export function useCheckedLegendItemsStatus(
+  data: PromiseHookState<
+    BarplotDataWithStatistics | HistogramDataWithCoverageStatistics | undefined
+  >,
```

Comment:
To avoid hardcoding different Viz types, I would use a type parameter:

```suggestion
export function useCheckedLegendItemsStatus<T>(
  data: PromiseHookState<T>,
```

Reply:
@jtlong3rd thank you for your comments. As Bob also suggested, I have changed to simply call the custom hook without assigning it as a const. 

As for the generic, yes I also thought of the case as, without the generic, I should have added all other data types for different plot Vizs 😄 Those are committed - will remove 'export data type' at barplot and histogram Vizs sooner or later.